x-pack/plugins/security_solution/public/common/components/agents/agent_status/agent_response_action_status.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useMemo } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiBadge, EuiFlexGroup, EuiFlexItem, EuiTextColor, EuiToolTip } from '@elastic/eui';
+import type { EndpointPendingActions } from '../../../../../common/endpoint/types';
+import type { ResponseActionsApiCommandNames } from '../../../../../common/endpoint/service/response_actions/constants';
+import { RESPONSE_ACTION_API_COMMAND_TO_CONSOLE_COMMAND_MAP } from '../../../../../common/endpoint/service/response_actions/constants';
+import { ISOLATING_LABEL, RELEASING_LABEL, ISOLATED_LABEL } from './endpoint/endpoint_agent_status';
+import { useTestIdGenerator } from '../../../../management/hooks/use_test_id_generator';
+
+const TOOLTIP_CONTENT_STYLES: React.CSSProperties = Object.freeze({ width: 150 });
+
+interface AgentResponseActionsStatusProps {
+  /** The host's individual pending action list as return by the pending action summary api */
+  pendingActions: EndpointPendingActions['pending_actions'];
+  /** Is host currently isolated */
+  isIsolated: boolean;
+  'data-test-subj'?: string;
+}
+
+export const AgentResponseActionsStatus = memo<AgentResponseActionsStatusProps>(
+  ({ pendingActions, isIsolated, 'data-test-subj': dataTestSubj }) => {
+    const getTestId = useTestIdGenerator(dataTestSubj);
+
+    interface PendingActionsState {
+      actionList: Array<{ label: string; count: number }>;
+      totalPending: number;
+      wasReleasing: boolean;
+      wasIsolating: boolean;
+      hasMultipleActionTypesPending: boolean;
+      hasPendingIsolate: boolean;
+      hasPendingUnIsolate: boolean;
+    }
+
+    const {
+      totalPending,
+      actionList,
+      wasReleasing,
+      wasIsolating,
+      hasMultipleActionTypesPending,
+      hasPendingIsolate,
+      hasPendingUnIsolate,
+    } = useMemo<PendingActionsState>(() => {
+      const list: Array<{ label: string; count: number }> = [];
+      let actionTotal = 0;
+      let actionTypesCount = 0;
+
+      Object.entries(pendingActions)
+        .sort()
```

Comment:
Q: why do we have to sort it first?

Reply:
So that we show pending actions in an alphabetical order.

Reply:
I wonder if we shouldn't show them in time order?

Reply:
Interesting idea. I'd suggest you create a UX ticket for this and we can follow up with the rest of the team. That being said, I think it is alright with alphabetical order or any deterministic sort order other than _time_ order. Using time to sort the actions will make the commands jump where an action is on the list and that would be quite annoying to be able to monitor when taking multiple actions. Besides, we show pending actions on a tooltip with the action name and the count of each pending, something like

```json5
get-file: 3
kill-process: 2
processes: 1
suspend-process: 3
```

**example**
![Screenshot 2024-05-13 at 16 06 40](https://github.com/elastic/kibana/assets/1849116/cc5e99dd-db63-43d9-9508-65732f453679)

Imagine this list being re-sorted with every action of these types or new types being created. Let's say you created a `release` or `upload` actions. The `release` will push  the last item (`suspend-process`) "down" and upload will push it "up". If you were to create a `processes` instead that would re-order the above list to show `processes: 2` at the top.

Reply:
Oh, no that's fine, I didn't realize it's this tooltip. Timestamp doesn't make sense in this situation :)